pkg/server/remote_storage_test.go

```diff
@@ -229,6 +229,63 @@ func TestInitializeRemoteStorageDownloadIdentifier(t *testing.T) {
 	require.Equal(t, []byte{1, 2, 3, 4, 5}, id)
 }
 
+func TestInitializeWithEmptyRemoteStorage(t *testing.T) {
+	dir := t.TempDir()
+
+	opts := DefaultOptions().WithDir(dir)
+	opts.RemoteStorageOptions.S3ExternalIdentifier = true
+
+	s := DefaultServer()
+
+	s.WithOptions(opts)
+
+	err := s.Initialize()
+	require.Error(t, err, ErrNoStorageForIdentifier)
+}
+
+func TestInitializeWithRemoteStorageWithoutIdentifier(t *testing.T) {
+	dir := t.TempDir()
+
+	opts := DefaultOptions().WithDir(dir)
+	opts.RemoteStorageOptions.S3ExternalIdentifier = true
+
+	s := DefaultServer()
+
+	s.WithOptions(opts)
+
+	m := memory.Open()
+	s.remoteStorage = m
+
+	err := s.initializeRemoteStorage(m)
+	require.Error(t, err, ErrNoRemoteIdentifier)
```

Comment:
```suggestion
	require.ErrorIs(t, err, ErrNoStorageForIdentifier)
```

Using `ErrorIs` to validate `err` is that specific type of error